compiler/rustc_builtin_macros/src/deriving/smart_ptr.rs

```diff
@@ -154,6 +154,23 @@ pub fn expand_deriving_smart_ptr(
     {
         let pointee = &mut impl_generics.params[pointee_param_idx];
         self_bounds = pointee.bounds.clone();
+        if !contains_maybe_sized_bound(&self_bounds)
+            && !contains_maybe_sized_bound_on_pointee(
+                &generics.where_clause.predicates,
+                pointee_ty_ident.name,
+            )
+        {
+            cx.dcx()
+                .struct_span_err(
+                    pointee_ty_ident.span,
+                    format!(
+                        "`SmartPointer` is meaningless because `{}` is not `?Sized`",
```

Comment:
I would probably phrase this as
```
"`derive(SmartPointer)` requires the use of `{}: ?Sized`"
```

Reply:
Or "`derive(SmartPointer)` requires `{}` is marked `?Sized`".

Reply:
I elected
```
`derive(SmartPointer)` requires {} to be marked `?Sized`
```